tests/integration-service/integration-with-env.go

```diff
@@ -0,0 +1,172 @@
+				}, 1*time.Minute, 1*time.Second).Should(Succeed(), "timeout when waiting for finalizer to be added")
+			})
+
+			It("waits for build PipelineRun to succeed", Label("integration-service"), func() {
+				Expect(pipelineRun.Annotations[snapshotAnnotation]).To(Equal(""))
+				Expect(f.AsKubeDeveloper.HasController.WaitForComponentPipelineToBeFinished(originalComponent, "",
+					f.AsKubeAdmin.TektonController, &has.RetryOptions{Retries: 2, Always: true}, pipelineRun)).To(Succeed())
+			})
+		})
+
+		When("the build pipelineRun run succeeded", func() {
+			It("checks if the BuildPipelineRun have the annotation of chains signed", func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForBuildPipelineRunToGetAnnotated(testNamespace, applicationName, componentName, chainsSignedAnnotation)).To(Succeed())
+			})
+
+			It("checks if the Snapshot is created", func() {
+				snapshot, err = f.AsKubeDeveloper.IntegrationController.WaitForSnapshotToGetCreated("", "", componentName, testNamespace)
+
+			})
+
+			It("checks if the Build PipelineRun got annotated with Snapshot name", func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForBuildPipelineRunToGetAnnotated(testNamespace, applicationName, componentName, snapshotAnnotation)).To(Succeed())
+			})
+
+			It("verifies that the finalizer has been removed from the build pipelinerun", func() {
+				timeout := "60s"
+				interval := "1s"
+				Eventually(func() error {
+					pipelineRun, err = f.AsKubeDeveloper.IntegrationController.GetBuildPipelineRun(componentName, applicationName, testNamespace, false, "")
+					Expect(err).ShouldNot(HaveOccurred())
+					if controllerutil.ContainsFinalizer(pipelineRun, pipelinerunFinalizerByIntegrationService) {
+						return fmt.Errorf("build pipelineRun %s/%s still contains the finalizer: %s", pipelineRun.GetNamespace(), pipelineRun.GetName(), pipelinerunFinalizerByIntegrationService)
+					}
+					return nil
+				}, timeout, interval).Should(Succeed(), "timeout when waiting for finalizer to be removed")
+			})
+
+			It("checks if CronJob spacerequest-cleaner exists", func() {
+				spaceRequestCleanerCronJob, err := utils.GetCronJobs()
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(strings.Contains(spaceRequestCleanerCronJob, "spacerequest-cleaner")).Should(BeTrue())
+			})
+
+			It("checks if all of the integrationPipelineRuns passed", Label("slow"), func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForAllIntegrationPipelinesToBeFinished(testNamespace, applicationName, snapshot)).To(Succeed())
+			})
+
+			It("checks if space request is created in namespace", func() {
+				spaceRequestsList, err := f.AsKubeAdmin.GitOpsController.GetSpaceRequests(testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(strings.Contains(spaceRequestsList.Items[0].GetName(), "task-spacerequest-")).Should(BeTrue())
+			})
+
+			It("checks if the passed status of integration test is reported in the Snapshot", func() {
```

Comment:
So there's a lot of code duplication from `tests/integration-service/integration.go` - have you thought about extending the existing test in `integration.go` instead?

I'm not familiar with integration-service tests very much, so I'm not sure if it'd be possible or more effective than having a separate test, just wondering.
cc @dheerajodha

Reply:
my idea of creating this in a separate test is the idea that this should be a short term solution , 
once we decide to drop it, so we delete this one with no dependencies of other tests.

Reply:
ok